src/Extension/Propagator/Jaeger/JaegerPropagator.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Extension\Propagator\Jaeger;
+
+use OpenTelemetry\API\Baggage\Baggage;
+use OpenTelemetry\API\Baggage\Entry; /** @phan-suppress-current-line PhanUnreferencedUseNormal */
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanContext;
+use OpenTelemetry\API\Trace\SpanContextInterface;
+use OpenTelemetry\API\Trace\SpanContextValidator;
+use OpenTelemetry\API\Trace\TraceFlags;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextInterface;
+use OpenTelemetry\Context\Propagation\ArrayAccessGetterSetter;
+use OpenTelemetry\Context\Propagation\PropagationGetterInterface;
+use OpenTelemetry\Context\Propagation\PropagationSetterInterface;
+use OpenTelemetry\Context\Propagation\TextMapPropagatorInterface;
+
+/**
+ * JaegerPropagator is a propagator that supports the specification for the header
+ * "uber-trace-id" used for trace context propagation across service boundaries.
+ * (https://www.jaegertracing.io/docs/1.52/client-libraries/#propagation-format)
+ */
+class JaegerPropagator implements TextMapPropagatorInterface
+{
+    private const UBER_TRACE_ID_HEADER = 'uber-trace-id';
+    private const UBER_BAGGAGE_HEADER_PREFIX = 'uberctx';
+
+    private const IS_NOT_SAMPLED = '0';
+    private const IS_SAMPLED = '1';
+    private const IS_DEBUG = '2';
+    private const DEFAULT_PARENT_SPAN_ID = '0';
+
+    private const FIELDS = [
+        self::UBER_TRACE_ID_HEADER,
+    ];
+
+    private static ?TextMapPropagatorInterface $instance = null;
+
+    public static function getInstance(): TextMapPropagatorInterface
+    {
+        if (self::$instance === null) {
+            self::$instance = new JaegerPropagator();
+        }
+
+        return self::$instance;
+    }
+
+    public function fields(): array
+    {
+        return self::FIELDS;
+    }
+
+    /** {@inheritdoc} */
+    public function inject(&$carrier, PropagationSetterInterface $setter = null, ContextInterface $context = null): void
+    {
+        $setter ??= ArrayAccessGetterSetter::getInstance();
+        $context ??= Context::getCurrent();
+        $spanContext = Span::fromContext($context)->getContext();
+
+        if (!$spanContext->isValid()) {
+            return;
```

Comment:
Baggage should still be propagated.
Might make sense to create a separate baggage propagator to allow users to enable/disable trace and baggage propagation separately.

Reply:
From what I understand, your suggestion would be to separate the propagators into something like, for example, `JaegerPropagator` and `JaegerBaggagePropagator`?

If you have another suggestion, could you give an example?